tools/esdlcomp/esdlcomp.h

```diff
@@ -605,6 +605,12 @@ class ParamInfo
         {
             out.appendf("\t\t<EsdlElement name='%s'", name);
             toStringXmlAttr(out);
+
+            if (xsdtype && *xsdtype && hidlMode)
```

Comment:
xsdtype is set for general strings that don't use an extended type.  This code results in:

		<EsdlElement name='PriorityDesc' type='string' hidlType='string'/>
		<EsdlElement name='PriorityLevel' type='int'/>
		<EsdlElement name='DateTimeScheduled' type='string' hidlType='dateTime'/>

Notice that the string type has both type and hidl_type equal to "string".

You could just check for that here.

if (xsdtype && *xsdtype && strcmp(xsdtype, "string")!=0)


Combining that with my other comments the output would look like:

		<EsdlElement name='PriorityDesc' type='string'/>
		<EsdlElement name='PriorityLevel' type='int'/>
		<EsdlElement name='DateTimeScheduled' type='string' xsd_extended_type='dateTime'/>